src/math_linalg/base_matrix.h

```diff
@@ -792,7 +792,7 @@ class base_matrix{
   ///< Assignment = copying (by value)
   void operator=(const base_matrix<T1>& ob){
 
-    if(this == &ob){  return *this;    }

```

Comment:
it should return *this - so that the comparison could be chained - this is as far as I understand the point of this return statement

Reply:
This function is marked as void, so, it cannot return anything

Reply:
Hmm... You are right, the another patch should be here